WWDC/VideoPlayerViewController.swift

```diff
@@ -177,15 +173,22 @@ final class VideoPlayerViewController: NSViewController {
 
     // MARK: - Player Observation
 
-    override func observeValue(forKeyPath keyPath: String?, of object: Any?, change: [NSKeyValueChangeKey: Any]?, context: UnsafeMutableRawPointer?) {
-        guard let keyPath = keyPath else { return }
+    private var playerStatusObserver: NSKeyValueObservation?
+    private var presentationSizeObserver: NSKeyValueObservation?
+    private var currentItemObserver: NSKeyValueObservation?
 
-        if keyPath == #keyPath(AVPlayer.currentItem.presentationSize) {
-            DispatchQueue.main.async(execute: playerItemPresentationSizeDidChange)
-        } else if keyPath == #keyPath(AVPlayer.status) {
-            DispatchQueue.main.async(execute: playerStatusDidChange)
-        } else {
-            super.observeValue(forKeyPath: keyPath, of: object, change: change, context: context)
+    private func setupPlayerObservers() {
+
+        playerStatusObserver = player.observe(\AVPlayer.status, options: [.initial, .new], changeHandler: { [weak self] (player, change) in
+            guard let strongSelf = self else { return }
```

Comment:
I've been using `guard let 'self' = self` (pretend the ' are backticks) in these cases lately, maybe we should standardize and use it throughout?

Reply:
I think I love this! Never considered it. Way less prone to people coming along and adding `self?.whatever` alongside the `strongSelf.otherthing` and getting the context all confused.

Reply:
Yeah and it leaves the syntax highlighting intact as well. Win/win/win 😄